src/initialization/InitElement.cpp

```diff
@@ -40,17 +40,23 @@ namespace impactx
             // Initialize the corresponding element according to its type
             if (element_type == "quad") {
                 amrex::Real ds, k;
+                int nslice = 1;
```

Comment:
We could consider adding a simulation parameter that can be used as the default value for all elements (unless otherwise specified). Would that make sense/be user friendly or is it better to force users to give us explicitly the substeps per element in any case?

Reply:
I suggest that we discuss this with Ji/others next week.  One issue is that elements have different lengths, and also different dynamics, so the optimal choice of step size may vary significantly from one element to the next.  This is why both IMPACT-Z and MaryLie/IMPACT input the number of substeps per element.  However, having a universal default could be useful for convergence studies.  One option is to replace the initialization "nslice = 1" above with "nslice = nsteps".

Reply:
yes, that's what I was thinking. Just a global non-`1` default that can be overwritten per element.

Reply:
I just added "num_steps" as an input, and this is now used as the default value of "nslice" for each element.